### ISSUE TITLE
ability to assign permissions to role by DAGs' Owner (additional to DAGs' name)

### ISSUE THREAD

**Issue by orsagiv94** [apache/airflow#23638]
### Description

When creating/updating Airflow role for a specific team,
when need to define his permissions for a set of DAGs whose relevant to that team, we'll have to add **manually** each relevant permission for each associated DAG's name ( = number of permissions X number of DAGs).

we need the ability to define role permissions based on DAG's Owner name which will reduce a lot of manual work for the Admin maintainers and will minimize the number of permissions defined on each role.

### Use case/motivation

In our organization, we are running Airflow as a workflow manager for about ~20 teams.
Each team manages its own DAGs, and we as Admins are the maintainers of all of their roles.

Every new DAG must be mentioned in his relevant team-based role twice (for read & edit permissions).

So in order to create a new role for team 'A' for example which has 10 DAGs, we'll have to create a role with at least 20 permissions which had to be defined manually.

in our organization, every DAG has an Owner defined on his DAG's python file.
Therefore, assigning permissions to a role by DAG's Owner instead of DAG's names will reduce significant manual maintenance work and be much more efficient. 

example - 
when setting a role, the permission - 
`read on DAG:tutorial` can be replaced with `read on OWNER:Team_A`

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by bugraoz93** [comment]
Hey @potiuk, would I take over this issue to do it? I am willing to work on it. Most probably, I will need help and more discussions over how we should do it. 
I am not %100 aware of the code base related to this one. Because of that, I will try to look at the code base that is responsible for this feature. Then I can come up with a solution to discuss.

**Comment by potiuk** [comment]
Sure. Best it to do exploration on your own and come up with some POC PR (even not complete) so that the help can be over the "real" code

**Comment by bugraoz93** [comment]
Sure, thanks a lot! That would be great.

**Comment by antoine-lefebvre** [comment]
Would be great to have this feature!

**Comment by bugraoz93** [comment]
I don't think this is possible with the written approach in the issue. There is [a draft AIP](https://cwiki.apache.org/confluence/x/eoqSEQ) (Airflow Improvement Proposal) for supporting multi-team deployment for Apache Airflow components, which is one of the candidates for Airflow v3. Stay tuned for this feature and upgrade your Apache Airflow version when it arrives. 

For now, I would suggest using `access_control` in the Dag with the combination of `dag_policy` where you can automate adding the Dags to certain roles regarding the Owner of the Dag.

I think we can even close this issue. It is not relevant anymore.

**Comment by bugraoz93** [comment]
This is handled from team-based deployment in 3.2.0 experimentally and will come with later versions as stable. Closing this in favour of https://cwiki.apache.org/confluence/display/AIRFLOW/AIP-67+Multi-team+deployment+of+Airflow+components
